fabtests/hpcs/src/test/generic.c

```diff
@@ -0,0 +1,256 @@
+ * ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#include <limits.h>
+#include <string.h>
+#include <stdio.h>
+#include <errno.h>
+#define __STDC_FORMAT_MACROS
+#include <inttypes.h>
+#include <unistd.h>
+
+#include <rdma/fi_errno.h>
+#include <rdma/fi_eq.h>
+
+#include <test/user.h>
+
+struct test_arguments {
+};
+
+void test_generic_free_arguments(struct test_arguments *arguments)
+{
+	free (arguments);
+}
+
+void test_generic_tx_init_buffer(const struct test_arguments *arguments,
+		uint8_t *buffer, size_t len)
+{
+	memset(buffer, UINT8_MAX, len);
+}
+
+void test_generic_rx_init_buffer(const struct test_arguments *arguments,
+		uint8_t *buffer, size_t len)
+{
+	memset(buffer, 0, len);
+}
+
+static struct fid_mr *test_generic_create_mr(const struct test_arguments *arguments,
+		struct fid_domain *domain, const uint64_t key,
+		uint8_t *buffer, size_t len, uint64_t access, uint64_t flags)
+{
+	int ret;
+	struct fid_mr *mr;
+
+	ret = fi_mr_reg(domain, buffer, len, access, 0, key, flags, &mr, NULL);
+	if (ret)
+		goto err_mr_reg;
+
+	return mr;
+
+err_mr_reg:
+	fprintf(stderr, "fi_mr_reg failed\n");
+	return NULL;
+}
+
+struct fid_mr *test_generic_tx_create_mr(const struct test_arguments *arguments,
+		struct fid_domain *domain, const uint64_t key,
+		uint8_t *buffer, size_t len, uint64_t access, uint64_t flags)
+{
+	return test_generic_create_mr(arguments, domain, key,
+			buffer, len, access, flags);
+}
+
+struct fid_mr *test_generic_rx_create_mr(const struct test_arguments *arguments,
+		struct fid_domain *domain, const uint64_t key,
+		uint8_t *buffer, size_t len, uint64_t access, uint64_t flags)
+{
+	return test_generic_create_mr(arguments, domain, key,
+			buffer, len, access, flags);
+}
+
+int test_generic_tx_destroy_mr(const struct test_arguments *arguments, struct fid_mr *mr)
+{
+	return fi_close(&mr->fid);
+}
+
+int test_generic_rx_destroy_mr(const struct test_arguments *arguments, struct fid_mr *mr)
+{
+	return fi_close(&mr->fid);
+}
+
+size_t test_generic_tx_window_usage(const struct test_arguments *arguments,
+		const size_t transfer_id, const size_t transfer_count)
+{
+	return 1;
+}
+
+size_t test_generic_rx_window_usage(const struct test_arguments *arguments,
+		const size_t transfer_id, const size_t transfer_count)
+{
+	return 1;
+}
+
+int test_generic_tx_transfer(const struct test_arguments *arguments,
+		const size_t transfer_id, const size_t transfer_count,
+		const fi_addr_t rx_address, struct fid_ep *endpoint,
+		struct op_context *op_context, uint8_t *buffer,
+		void *desc, uint64_t key, int rank, uint64_t flags)
+{
+	return 0;
+}
+
+int test_generic_rx_transfer(const struct test_arguments *arguments,
+		const size_t transfer_id, const size_t transfer_count,
+		const fi_addr_t tx_address, struct fid_ep *endpoint,
+		struct op_context *op_context, uint8_t *buffer,
+		void *desc, uint64_t flags)
+{
+	return 0;
+}
```

Comment:
Fun fact - when we have functions like this, we know we're doing something wrong!